services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -314,6 +319,35 @@ private boolean isDuplicateKeyError(final Throwable throwable) {
                                 Optional.ofNullable(deviceDto.getVersion()))));
     }
 
+    private Future<JsonArray> processResolveGroupMembers(final String tenantId, final JsonArray viaGroups,
+            final Span span) {
+        final JsonObject resolveGroupMembersQuery = new MongoDbDocumentBuilder()
+                .withTenantId(tenantId).document()
+                .put(String.format("%s.%s", MongoDbDeviceRegistryUtils.FIELD_DEVICE,
+                        RegistryManagementConstants.FIELD_MEMBER_OF),
+                        new JsonObject()
+                                .put("$exists", true)
+                                .put("$in", viaGroups));
+        //Retrieve only the deviceId instead of the whole document.
+        final FindOptions findOptionsForDeviceId = new FindOptions()
+                .setFields(new JsonObject().put(RegistrationConstants.FIELD_PAYLOAD_DEVICE_ID, true).put("_id", false));
+        final Promise<List<JsonObject>> resolveGroupMembersPromise = Promise.promise();
+
+        mongoClient.findWithOptions(config.getCollectionName(), resolveGroupMembersQuery, findOptionsForDeviceId,
+                resolveGroupMembersPromise);
+
+        return resolveGroupMembersPromise.future()
+                .map(deviceIdsList -> {
+                    final JsonArray deviceIds = Optional.ofNullable(deviceIdsList)
+                            .map(ok -> deviceIdsList.stream()
+                                    .map(json -> json.getString(RegistrationConstants.FIELD_PAYLOAD_DEVICE_ID))
+                                    .collect(Collectors.collectingAndThen(Collectors.toList(), JsonArray::new)))
+                            .orElse(new JsonArray());
+                    span.log(String.format("successfully resolved group members for the tenant [%s]", tenantId));
```

Comment:
The span should already be tagged with the tenant ID.

Reply:
Since the tenant ID is already being tagged, now I have removed logging the tenant ID and pushed the changes.